### PR TITLE
Bp 852 woocommerce logging

### PR DESCRIPTION
**Pull request by harli91**

### CODE REVIEW

File: library/Buckaroo_Logger.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * @category  Payment_Gateways
+ * @package   Buckaroo
+ * @author    Buckaroo <support@buckaroo.nl>
+ * @copyright 2021 Copyright (c) Buckaroo B.V.
+ * @license   MIT https://tldrlegal.com/license/mit-license
+ * @version   GIT: 2.25.0
+ * @link      https://www.buckaroo.eu/
+ */
+if (!class_exists('Buckaroo_Logger_Storage'))
```
Comment (vladislav-padalka-hys) [inline library/Buckaroo_Logger.php:16]: @harli91 can you please clarify why we need that check in case if we include it by require_once only ?

**Comment by vladislav-padalka-hys** [comment]
@harli91 I see no sense anymore to keep not used constants for logging like : 
    const LOG = true;
    const LOG_DIR = '/log/';

**Comment by vladislav-padalka-hys** [comment]
@harli91 you've changed format for files to be all in one file (in opposite to per day basis used previously, like api-response-log-2022-01-17.txt) 
I see more lacks in this than benefits and nobody has requested this. 
Can you please return to previous scheme ? 
It will simplify a lot cleaning of expired logs also

File: templates/Buckaroo_Report_Page.php [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+            return 0;
+        }
+        $file = new \SplFileObject($this->get_file_path(), 'r');
+        $file->setFlags(SplFileObject::DROP_NEW_LINE);
+        $lines = [];
+
+        while (!$file->eof()) {
+            $rowData = $file->current();
+            $key = $file->key();
+            if (str_contains($rowData, "-->")) {
```
Comment (vladislav-padalka-hys) [inline templates/Buckaroo_Report_Page.php:287]: @harli91 Fatal error: Uncaught Error: Call to undefined function str_contains() in /home/vlad/workspace/buckaroo_wp/wp-content/plugins/wc-buckaroo-bpe-gateway/templates/Buckaroo_Report_Page.php:287

function is for PHP 8 , while I've tested on 7.4 and it should be compatible with it

**Comment by harli91** [comment]
@vladislav-padalka-hys  thanks for the feedback, i will look into what you sad

**Comment by vladislav-padalka-hys** [comment]
@harli91 sorry I don't see new commits to review...did I miss something ?

**Comment by harli91** [comment]
sorry about that, i update the file system to create a file for each day and append logs to that, you can check the updated code

**Comment by vladislav-padalka-hys** [comment]
thank you for your changes - looks good !
last thing if possible : could you please add into "Report" module ability to download specific log as file , because if log file is big enough then it's not useful to open it in textarea and copy-paste ?

**Comment by vladislav-padalka-hys** [comment]
@harli91 and to separate requests from each other in logging could you please add kind of process id (uniqid() or getmypid()) to each row as it's in m2. Example : 
2022-01-18 08:38:23|||Start Response|||
->
2022-01-18 08:38:23|||61e5720e292af|||Start Response|||

**Comment by harli91** [comment]
added download link & unique id to the file logger

File: library/Buckaroo_Logger_Storage.php [hunk 3]
```diff
@@ -109,8 +109,16 @@ protected function store_in_all(array $info)
...
     {
         @file_put_contents(
-            $this->get_file_storage_location().date('d-m-Y').".log",
-            implode("|||", $info) . PHP_EOL,
+            self::get_file_storage_location().date('d-m-Y').".log",
+            implode(
+                "|||",
+                [
+                    $info[0],
+                    uniqid("", true),
```
Comment (vladislav-padalka-hys) [inline library/Buckaroo_Logger_Storage.php:117]: Hm...this identifier is always different now, even within scope of the same HTTP request 
Check below : 

2022-01-18 11:55:53|||61e6aac977b0e6.21302357|||Order status: processing|||
2022-01-18 11:55:53|||61e6aac97977a3.82146698|||Response order status: completed|||
2022-01-18 11:55:53|||61e6aac97b3e70.59053511|||Status message: Transaction successfully processed|||
2022-01-18 11:55:53|||61e6aac97cfba2.53948149|||Order already in final state or  have the same status as response. Order status: processing|||

So it should be something like below to be the same for the same HTTP request: 

if (empty(self::$processUid)) {
  self::$processUid = uniqid();
}

P.S.: add it also to DB logger

**Pull request merged**